app/background.cc

```diff
@@ -11,23 +11,24 @@
 
 // Internal state of background processing
 struct background_t {
-    int dbg;
-    bool vid;
+    int debug;
+    bool video;
     volatile bool run;
     cv::VideoCapture cap;
-    int frm;
+    int frame;
     double fps;
     cv::Mat raw;
-    std::thread thr;
-    std::mutex mux;
+    cv::Mat thumb;
+    std::thread thread;
+    std::mutex rawmux;
+    std::mutex thumbmux;
```

Comment:
```suggestion
    std::mutex rawmux;
    cv::Mat thumb;
    std::mutex thumbmux;
    std::thread thread;
```

Reordering to have all mutexes and cv::Mat before the thread object (also determines initialization order).

Maybe rename `thread` to `segmentation_thread`?

Reply:
will do.